### PR TITLE
Ensure that the context manager is exited

### PR DESCRIPTION
**Pull request by pabloferz**
For some reason the SystemView context manager was not always being exited. This seems to work and it should be safe as we are now returning copies of the arrays in `take_snapshot`.

As far as I can tell this fixes the segfault reported in #189 (both jax and hoomd-dlext were trying to free the arrays from the initial snapshot).

### CODE REVIEW

File: pysages/backends/hoomd.py [hunk 1]
```diff
@@ -107,11 +107,11 @@ def is_on_gpu(context):
 def take_snapshot(wrapped_context, location=default_location()):
     context = wrapped_context.context
     sysview = wrapped_context.view
-    positions = asarray(positions_types(sysview, location, AccessMode.Read))
-    vel_mass = asarray(velocities_masses(sysview, location, AccessMode.Read))
-    forces = asarray(net_forces(sysview, location, AccessMode.ReadWrite))
-    ids = asarray(rtags(sysview, location, AccessMode.Read))
-    imgs = asarray(images(sysview, location, AccessMode.Read))
+    positions = copy(asarray(positions_types(sysview, location, AccessMode.Read)))
```
Comment (InnocentBug) [inline pysages/backends/hoomd.py:110]: Is that OK.
What are we copying here.
Just the pointer to the data?
The context manager for the data?
Or a deep copy?

File: pysages/backends/hoomd.py [hunk 2]
```diff
@@ -197,20 +197,20 @@ def bind(
...
+    wrapped_context.view = sysview
+    wrapped_context.run = hoomd.run
     helpers, restore, bias = build_helpers(context, sampling_method)
 
-    with SystemView(context.system_definition) as sysview:
-        wrapped_context.view = sysview
-        wrapped_context.run = hoomd.run
-
+    with sysview:
         snapshot = take_snapshot(wrapped_context)
```
Comment (InnocentBug) [inline pysages/backends/hoomd.py:206]: So, the only function we call inside the context manager is the snapshot.
But don't we lose the context manager access for the snapshot once we leave the sysview context?

Seems to me like we use a dangling pointer here.

File: pysages/backends/hoomd.py (hunk 1 above)
Comment (pabloferz) [reply pysages/backends/hoomd.py:110]: This copies the whole array. But this is only called once at the very beginning of the run (this is different than the one called internally in `Sampler`). But I think I can get it to work without calling `copy` here.

Comment (pabloferz) [reply pysages/backends/hoomd.py:110]: Solving this differently is going to be trickier than I thought, but this should work fine for now. I'd say we move ahead and find a better way to fix it later.

File: pysages/backends/hoomd.py (hunk 2 above)
Comment (pabloferz) [reply pysages/backends/hoomd.py:206]: We won't because, we would be returning copies of all arrays.

**Review by InnocentBug — APPROVED** [review]
Ok, that is fine with me for now.

**Pull request merged**